phac-nml/irida-next
# Update: Updated UI for transfering a project namespace

## What does this PR do and why?

Moves transferring a project to a different namespace into it's own card container and introducing an "Advanced settings" section within the edit page.

## Screenshots or screen recordings

Updated project settings panel:

![](https://github.com/phac-nml/irida-next/assets/11295750/58b223aa-4dc9-48bc-bac9-3f47e4bc8e63)

Updated confirmation dialog:

![](https://github.com/phac-nml/irida-next/assets/11295750/09334e12-57fe-4c2d-be67-46c5ae696967)

## How to set up and validate locally

1.  Starting with a fresh development database, login with user1
2.  Go to the settings page for [**Streptococcus / Streptococcus pyogenes / Outbreak 2022**](http://localhost:3000/streptococcus/streptococcus-pyogenes/outbreak-2022/-/edit)
3.  You will see the updated page with a new "Advanced settings" > "Transfer Project"
4.  The "Transfer project" button should be disabled
5.  From the "Select a new namespace" select input, select `S`treptococcus / Streptococcus agalactiae"
6.  The "Transfer project" button should become enabled, click on it.
7.  A confirm dialog should be displayed where you need to type the project name, enter "Outbreak 2022" and click "Confirm"
8.  The project cannot be transferred to the namespace and an error message should be displayed under the select input.
9.  From the "Select a new namespace" select, select "user1@email.com", and click the "Transfer project" button and repeat the confirmation.
10.  This transfer works and the user is redierected back to the main project page.  If you look in the breadcrumbs, you will see that the namespace has been updated (also in the url).

![](https://github.com/phac-nml/irida-next/assets/11295750/13a184b4-4814-4f72-82b7-782748b9d1be)
